mytardis_rocrate_builder/rocrate_builder.py

```diff
@@ -154,47 +209,86 @@ def __add_person_to_crate(self, person: Person) -> ROPerson:
+        # for identifier in person.mt_identifiers:
+        #     if _ := orcid_regex.fullmatch(identifier):
+        #         person_id = identifier
         # If no orcid is found check for UPIs
-        if not person_id:
-            for identifier in person.identifiers:
+        if not upi_regex.fullmatch(str(person_id)):
+            for identifier in person.mt_identifiers:
                 if _ := upi_regex.fullmatch(identifier):
                     person_id = identifier
         # Finally, if no orcid or UPI is found, use the first identifier
-        if not person_id:
-            person_id = person.identifiers[0]
+        # if not person_id:
+        #     person_id = person.mt_identifiers[0]
+        person_obj = ROPerson(
+            self.crate,
+            person_id,
+            properties={"name": person.name, "email": person.email},
+        )
+
         if not any(
             (
                 entity.type in ["Organization", "ResearchOrganization"]
                 and entity.id == person.affiliation.id
             )
             for entity in self.crate.get_entities()
         ):
-            self.__add_organisation(person.affiliation)
-        person_obj = ROPerson(
-            self.crate,
-            person_id,
-            properties={
-                "name": person.name,
-                "email": person.email,
-                "affiliation": person.affiliation.id,
-            },
-        )
-        if len(person.identifiers) > 1:
-            for identifier in person.identifiers:
+            person_obj.append_to(
+                "affiliation", self.__add_organisation(person.affiliation)
+            )
+        if len(person.mt_identifiers) > 1:
+            for identifier in person.mt_identifiers:
                 if identifier != person_id:
                     person_obj.append_to("identifier", identifier)
         self.crate.add(person_obj)
         return person_obj
 
+    def add_user(self, user: User) -> ROPerson:
+        """Add a mytardis django user to the crate,
+        including adding this user as a person.
+
+        Args:
+            user (User): the user to be added
+
+        Returns:
+            ROPerson: the user as an RO-Crate entity
+        """
+        user_entity: ROPerson = self.crate.dereference(
+            user.roc_id
+        ) or self.__add_person_to_crate(user)
+        if user.groups:
+            for group in user.groups:
+                ro_group = self.crate.dereference(group.roc_id) or self.add_group(group)
+                ro_group.append_to("has_member", user_entity)
+                user_entity.append_to("groups", ro_group)
+        self._add_optional_attr(user_entity, "permissions", user.permissions)
+        self._add_optional_attr(
+            user_entity, "isDjangoAccount", user.is_django_account, True
+        )
+        self._add_optional_attr(user_entity, "is_staff", user.is_staff, True)
+        self._add_optional_attr(
```

Comment:
Part of me wonders if this is a good idea. As admins we can regenerate people's passwords and with SSO it is not necessary to hold the hashed password in the RO Crate. I suggest we strip this out.